### ISSUE TITLE
Allow to override SnippetChooserBlock.icon

### ISSUE THREAD

**Issue by mnieber** [wagtail/wagtail#4040]
Wagtail version: 1.13

I have a TeamMember snippet that I am trying to show with a "group" icon in the admin. It seems this is not supported, as I am always getting the "snippet" icon

![selection_060](https://user-images.githubusercontent.com/818892/32979955-afbb94be-cc5e-11e7-93dc-e4bdedb13a67.png)

    class TeamBlock(blocks.StructBlock):
        team_members = blocks.StreamBlock([
            (
                'teamMember',
                blocks.SnippetChooserBlock(TeamMember, icon='group')
            )
        ])

For completeness, this is my snippet class (by the way, in case you are wondering: I get all my imports from a custom `blocks` module which imports various standard wagtail classes): 

    @blocks.register_snippet
    class TeamMember(models.Model):
        name = models.CharField(max_length=255)
        image = blocks.wagtail_image_field()
        text = blocks.RichTextField()

        panels = [
            blocks.FieldPanel('name'),
            blocks.ImageChooserPanel('image'),
            blocks.FieldPanel('text'),
        ]

        def __str__(self):  # noqa
            return self.name

**Comment by lb-** [comment]
@mnieber thank you for submitting this idea.

The issue of snippet icons might need to be looked at in a broader scale as they are hard coded to use the same icon throughout a lot of code.

---

An interim solution would be just do insert some custom css by doing the following:

1. Pass in a `classname` argument to your `SnippetChooserBlock`

`('team_member', SnippetChooserBlock('base.people', classname='snippet-chooser-icon-group')),`

This feature is mentioned in the docs for StructBlock and classname is used in various examples in the [streamfield docs](http://docs.wagtail.io/en/v1.13.1/topics/streamfield.html#custom-editing-interfaces-for-structblock).

2. Create a css file in `static/css/snippet_chooser_block.css` file with the following
```
.snippet-chooser-icon-group .snippet-chooser .unchosen:before, .snippet-chooser .chosen:before { content: 'O'; /*group icon*/ }
```
This will add the icon with the character `O`, which translates to the group icon. You can see how the letters map to icons [here](https://github.com/wagtail/wagtail/blob/master/wagtail/wagtailadmin/static_src/wagtailadmin/scss/_variables-icons.scss), if you wanted to use Saas you could use the icon name from these variables instead.

3. Add a hook in `wagtail_hooks.py` to insert the css as per the [docs](http://docs.wagtail.io/en/v1.13.1/reference/hooks.html#insert-editor-css)
```
from wagtail.wagtailcore import hooks
from django.contrib.staticfiles.templatetags.staticfiles import static
from django.utils.html import format_html
# inert editor css to add custom snippet chooser block icons
@hooks.register('insert_editor_css')
def editor_css():
    return format_html(
        '<link rel="stylesheet" href="{}">',
        static('/css/snippet_chooser_block.css')
    )
```

**Comment by lb-** [comment]
Additional notes:

* When implementing, see if we want to enable custom snippet icons via `register_snippet` decorator or potentially honour the `class Meta: icon = 'something'` in the model definition that is registered as a snippet.
* See also #1224 & #2349 - Icon improvements (svg, not using 'letters' when making icons) we want to ensure it easy to add new icons and work with the existing icon set wherever possible.

**Comment by gasman** [comment]
Fixed in #10257
